File: test_api.js
const readline = require('readline');
const fetch = require('node-fetch');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Escribe el código del producto a buscar: ', (codigo) => {
    const url = `http://localhost/apis/buscar_producto.php?codigo=${encodeURIComponent(codigo)}`;
    console.log(`Consultando: ${url}`);

    fetch(url)
        .then(res => {
            console.log(`Código de respuesta HTTP: ${res.status}`);
            return res.json();
        })
        .then(data => {
            console.log("Respuesta del servidor:");
            console.log(JSON.stringify(data, null, 2));
        })
        .catch(err => {
            console.error("Error al conectarse con la API:");
            console.error(err);
        })
        .finally(() => rl.close());
});


//Ejecutar el script
// node test_api.js

File: test_api.py
import requests

def main():
    codigo = input("Escribe el código del producto a buscar: ").strip()

    url = f"http://localhost/apis/buscar_producto.php?codigo={codigo}"
    print(f"Consultando: {url}")

    try:
        response = requests.get(url)
        print(f"Código de respuesta HTTP: {response.status_code}")
        print("Respuesta del servidor:")
        print(response.json())
    except requests.exceptions.RequestException as e:
        print("Error al conectarse con la API:")
        print(e)
    except ValueError:
        print("La respuesta no es un JSON válido.")

if __name__ == "__main__":
    main()
# ejecutar el script
# python test_api.py